### ISSUE TITLE
Fix test file generation errors when using latest ncgen

### ISSUE THREAD

**Issue by benjwadams** [ioos/compliance-checker#541]
Currently, most test data is generated from CDL files outputting to NetCDF files.  It appears that newer versions of `ncgen` bundled with the NetCDF 4.5.0 library will fail to generate a number of the files which were generated specifically for testing.
E.g. this snippet from `tests/data/bad_missing_data.cdl`, which contains intentionally problematic data will cause ncgen to fail in newer versions.

```
int64 temp(time, time) ;
   temp:_FillValue = -999L ;
   temp:valid_min = 0L ;
   temp:valid_max = 10L ;
   temp:units = "K" ;
```

Fixing this would involve reworking the files to generate properly under all newer versions of ncgen, and modifying test conditions where necessary.

**Comment by kwilcox** [comment]
`libnetcdf 4.4` is the most recent availabe on `conda-forge` so I can't test or verify this.... can you try creating using the a python library and `libnetcdf 4.5`? I'm curios if its an `ncgen` regression/feature or an issue with the acutal C library. The below works in `4.4`.

```python
In [1]: import netCDF4 as nc4

In [2]: with nc4.Dataset('testdims.nc', mode='w') as ncd:
   ...:     ncd.createDimension('time', 20)
   ...:     ncd.createVariable('foo', int, ('time', 'time'))
```

**Comment by benjwadams** [comment]
Closing due to fix in #544.
